### PR TITLE
fix: Update `getBlockByHash` and `getBlockByNumber` to pass `timestamp` to MAPI calls

### LINKED ISSUE #3040: Update `eth_getBlockByHash` and `eth_getBlockByNumber` to pass a timestamp range in mirror node queries
In the `eth_getBlockByHash` and `eth_getBlockByNumber` methods, ensure the existing `timestampRangeParams` are used for subsequent API calls to `getLogsWithParams` and `resolveEvmAddress`. This change is necessary to handle the new 60-day limit on querying historical data in the mirror node.

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

- adds improvements to `MirrorNodeClient`:
- enhanced `getAccount` method to optionally accept a `ITimestampParams` parameter
- enhanced `prepareLogsParams`, `getContractResultsLogs` and `getContractResultsLogsByAddress` with strong typing on the `params` object that they accept - it now uses the `IContractLogsResultsParams` interface and this helps us track exactly where those methods are called and with which params passed to them
- updates the `eth_getBlockByHash` and `eth_getBlockByNumber` to pass a timestamp range from the fetched block to subsequent calls to `getLogsWithParams`

**Related issue(s)**:

Fixes #3040

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ebadiere — APPROVED** [review]
LG.

File: packages/relay/tests/lib/eth/eth_getBlockByHash.spec.ts [hunk 1]
```diff
@@ -80,7 +84,7 @@ describe('@ethGetBlockByHash using MirrorNode', async function () {
     sdkClientStub = sinon.createStubInstance(SDKClient);
     getSdkClientStub = sinon.stub(hapiServiceInstance, 'getSDKClient').returns(sdkClientStub);
     restMock.onGet('network/fees').reply(200, DEFAULT_NETWORK_FEES);
-    restMock.onGet(ACCOUNT_WITHOUT_TRANSACTIONS).reply(200, MOCK_ACCOUNT_WITHOUT_TRANSACTIONS);
+    restMock.onGet(`accounts/${LONG_ZERO_ADDRESS}?transactions=false`).reply(200, MOCK_ACCOUNT_WITHOUT_TRANSACTIONS);
```
Comment (konstantinabl) [inline packages/relay/tests/lib/eth/eth_getBlockByHash.spec.ts:87]: why removing the constant, you think it is unecessary or theres another reason?

**Review by Nana-EC — DISMISSED** [review]
LG but please update PR description for reviewers

**Comment by victor-yanev** [comment]
After discussing this with xin, it seems like those changes were not needed

**Pull request closed without merging**